packages/runtime/container-runtime/src/summary/runningSummarizer.ts

```diff
@@ -648,13 +648,21 @@ export class RunningSummarizer implements IDisposable {
 						return;
 					}
 
-					// Check for retryDelay that can come from summaryNack or upload summary flow.
-					// Retry the same step only once per retryAfter response.
 					overrideDelaySeconds = result.retryAfterSeconds;
-					if (overrideDelaySeconds === undefined || summaryAttemptsPerPhase > 1) {
+
+					// There are couple of scenarios where we want to retry an attempt with the same options:
+					// 1. If the server asked to retry after a delay and this is the first attempt in a phase. This can happen
+					// due to throttling during summary upload or we got a summaryNack.
+					// 2. If container runtime hits transient failures that may be fixed on a retry. For example, failures such
+					// as local data stores created during summarize can be fixed on retry. This should never happen in summarizer
+					// but unfortunately it does today.
+					const retrySameAttempt =
```

Comment:
Container Runtime can use `retryAfterSeconds` in the error returned to control if it thinks that a retry of summarize attempt might fix problems. In this PR, summarizer node failures have this behavior but it can be extended other kinds of failures if needed.